hooks/tk-multi-loader2/tk-houdini_actions.py

```diff
@@ -15,11 +15,82 @@
 import re
 import sgtk
 import sys
+import hou
 
 HookBaseClass = sgtk.get_hook_baseclass()
 
 class CustomHoudiniActions(HookBaseClass):
 
+    def generate_actions(self, sg_publish_data, actions, ui_area):
+        """
+        Returns a list of action instances for a particular publish.
+        This method is called each time a user clicks a publish somewhere in the UI.
+        The data returned from this hook will be used to populate the actions menu for a publish.
+
+        The mapping between Publish types and actions are kept in a different place
+        (in the configuration) so at the point when this hook is called, the loader app
+        has already established *which* actions are appropriate for this object.
+
+        The hook should return at least one action for each item passed in via the
+        actions parameter.
+
+        This method needs to return detailed data for those actions, in the form of a list
+        of dictionaries, each with name, params, caption and description keys.
+
+        Because you are operating on a particular publish, you may tailor the output
+        (caption, tooltip etc) to contain custom information suitable for this publish.
+
+        The ui_area parameter is a string and indicates where the publish is to be shown.
+        - If it will be shown in the main browsing area, "main" is passed.
+        - If it will be shown in the details area, "details" is passed.
+        - If it will be shown in the history area, "history" is passed.
+
+        Please note that it is perfectly possible to create more than one action "instance" for
+        an action! You can for example do scene introspection - if the action passed in
+        is "character_attachment" you may for example scan the scene, figure out all the nodes
+        where this object can be attached and return a list of action instances:
+        "attach to left hand", "attach to right hand" etc. In this case, when more than
+        one object is returned for an action, use the params key to pass additional
+        data into the run_action hook.
+
+        :param sg_publish_data: Shotgun data dictionary with all the standard publish fields.
+        :param actions: List of action strings which have been defined in the app configuration.
+        :param ui_area: String denoting the UI Area (see above).
+        :returns List of dictionaries, each with keys name, params, caption and description
+        """
+
+        # get the existing action instances
+        action_instances = super(CustomHoudiniActions, self).generate_actions(sg_publish_data, actions, ui_area)
+
+        if "image_plane" in actions:
+            action_instances.append({
+                "name": "image_plane",
+                "params": None,
+                "caption": "Create Image Plane",
+                "description": "Creates an image plane for the selected item.."
```

Comment:
Should we mention for a camera here?
(To distinguish from free image plane. Thinking of Maya...I don't even know if such a thing is possible in Houdini.